### PR TITLE
SELinux fixes for master/3.20.0

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8092/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8092/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8092/

File: misc/selinux/cfengine-enterprise.te [hunk 1]
```diff
@@ -614,6 +614,52 @@ allow cfengine_httpd_t shell_exec_t:file map;
 allow cfengine_httpd_t shell_exec_t:file { execute execute_no_trans };
 
 
+#============= cfengine_apachectl_t ==============
```
Comment (nickanderson) [inline misc/selinux/cfengine-enterprise.te:619]: Only because i still feel like a SELinux newb, i kind of want a comment for each one of the allow lines explaining what it does in plain english but that seems like overkill and I can kind of follow.

File: misc/selinux/cfengine-enterprise.te [hunk 2]
```diff
@@ -811,3 +811,17 @@ allow cfengine_cfbs_t sssd_var_lib_t:sock_file write;
...
+
+
+#============= TO REMOVE ==============
+# Daemons should have proper access to files based on DAC rules (file
+# permissions), not just because they run under root.
+dontaudit cfengine_execd_t self:capability dac_override;
+dontaudit cfengine_serverd_t self:capability { dac_override dac_read_search };
+
+# cf-promises run by the daemons shouldn't check if a function is available in
+# PostgreSQL, the respective returnszero() is in an agent bundle
```
Comment (nickanderson) [inline misc/selinux/cfengine-enterprise.te:823]: ack, more reasons to kill off the run of cf-promises by each component.

**Review by nickanderson — APPROVED** [review]

File: misc/selinux/cfengine-enterprise.te (hunk 1 above)
Comment (vpodzime) [reply misc/selinux/cfengine-enterprise.te:619]: Yeah... Reasons behind some of those lines are quite unclear and TBH, I didn't spend hours figuring out why we need every single one of them. Things like `sssd_t` come from NSS plugins and there are many other actions happening behind the scenes.

**Comment by vpodzime** [comment]
Merging this because we need backports of some parts for the LTS releases. @olehermanse feel free to suggest follow-ups.
